src/SignalR/common/SignalR.Common/test/Internal/Protocol/MessagePackHubProtocolTests.cs

```diff
@@ -26,10 +26,15 @@ public void SerializerCanSerializeTypesWithNoDefaultCtor()
             AssertMessages(new byte[] { ArrayBytes(5), 3, 0x80, StringBytes(1), (byte)'0', 0x03, ArrayBytes(1), 42 }, result);
         }
 
-        [Fact]
-        public void WriteAndParseDateTimeConvertsToUTC()
+        [Theory]
+        [InlineData(DateTimeKind.Utc)]
+        [InlineData(DateTimeKind.Local)]
+        [InlineData(DateTimeKind.Unspecified)]
+        public void WriteAndParseDateTimeConvertsToUTC(DateTimeKind dateTimeKind)
         {
-            var dateTime = new DateTime(2018, 4, 9);
+            // The messagepack Timestamp format are always deserialized as UTC
+            // calling ToUniversalTime() before serialization ensure that both input and output will have the same DateTimeKind : DateTimeKind.Utc
+            var dateTime = new DateTime(2018, 4, 9, 0, 0, 0, dateTimeKind).ToUniversalTime();
```

Comment:
This defeats any interesting testing of the msgpack library from your theory though. You're *always* passing UTC time into the serializer, so why bother with local and Unspecified?

If you wanted to test behavior of Local and Unspecified, you'd need to actually serialize those values.

Reply:
yep,
So just UTC, or convertion right before assertion ?

I'm not sure what to test here, I don't want to write tests that might already be within `MessagePack` repo

Reply:
Well I don't consider that you're testing the messagepack library here, but rather the `HubProtocol` front-end for it. So test whatever makes sense for that front-end.

Reply:
will take another shoot after a good sleep ;) thx again for the reviews

Reply:
Sorry about that yesterday ;)
I've change a bit the Theory, is that more acceptable ?